lib/fastly_nsq/rake_task.rb

```diff
@@ -44,7 +44,11 @@ def run_tasks
         end
       end
 
-      listener_threads.map(&:join)
+      while listener_threads.any?(&:status) do
+        listener_threads.each do |thread|
+          thread.join(1)
+        end
+      end
```

Comment:
With the addition of a time (one second here), `Thread#join` will return to the main thread after that time regardless of what happens in the sub thread. The inner `.each` will loop over all threads once, and then be done taking at most `threads.count` seconds.
The check of `Thread#status` keeps the while loop looping while there are still threads that are available to be `#join`'ed

The case where `.any?` is reached is a good point. I can change this to a `begin` `while` loop so that the join's are executed at least once.